src/main/kotlin/stringaddcalculator/domain/StringTokenizer.kt

```diff
@@ -0,0 +1,24 @@
+package stringaddcalculator.domain
+
+class StringTokenizer(
+    private val separators: Separators = Separators()
+) {
+    fun tokens(input: String): List<String> {
+        val (customSeparator, source) = extract(input)
+        if (customSeparator != null) {
+            separators.add(customSeparator)
+        }
+        return source.split(separators.toRegex())
+    }
+
+    private fun extract(input: String): Pair<String?, String> {
+        val matchResult = CUSTOM_REGEX.find(input)
+        return matchResult?.let {
+            it.groupValues[1] to it.groupValues[2]
+        } ?: (null to input)
```

Comment:
```suggestion
        return if (matchResult != null) {
            matchResult.groupValues[1] to matchResult.groupValues[2]
        } else {
            (null to input)
        }
```

expression body, safe call, elvis operator에 대해서 사용을 자주 하는 것은 좋지 않다고 생각해요!

`?.` ~ `?:` 를 같이 사용하게 되면, 자칫 잘못하면 `?.` 에 있는 경우와 `?:`에 있는 경우 모두 실행이 될 수 있기 때문에도 있습니다.

```kotlin

return test?. {
        println("null 1")
        null
} ?: println("null 2")

// result
// null 1
// null 2
```

그리고 null에 대해서 명확할 때는 if 문을 통하여 표현해 주는 게 더 가독성 있어 보이지 않을까요? 🤔

Reply:
확실히 이전 코드는 저도 작성하면서 이게 맞나? 싶은 코드였어요🥲

리뷰어님 의견처럼 if 문으로 null을 명시적으로 확인하는 것도 가독성 있어 보이는군요!

저도 조금 고민을 해보았는데, 위 로직은 elvis 연산자만 활용해도 가독성 있는 코드가 될 것 같네요🤔
```kotlin
val matchResult = CUSTOM_REGEX.find(input) 
            ?: return null to input
return matchResult.groupValues[1] to matchResult.groupValues[2]
```